src/models/polar.jl

```diff
@@ -0,0 +1,509 @@
+# Polar formulation
+#
+struct PolarForm{T, VT, AT} <: AbstractFormulation where {T, VT, AT}
+    network::PS.PowerNetwork
+    device::Device
+    x_min::VT
+    x_max::VT
+    u_min::VT
+    u_max::VT
+    # costs
+    costs_coefficients::AT
+    # Constants
+    active_load::VT
+    reactive_load::VT
+    # struct
+    ybus_re::Spmat
+    ybus_im::Spmat
+    AT::Type
+end
+
+function PolarForm(pf::PS.PowerNetwork, device)
+    if isa(device, CPU)
+        VT = Vector
+        M = SparseMatrixCSC
+        AT = Array
+    elseif isa(device, CUDADevice)
+        VT = CuVector
+        M = CuSparseMatrixCSR
+        AT = CuArray
+    end
+    ybus_re, ybus_im = Spmat{VT}(pf.Ybus)
+    coefs = PS.get_costs_coefficients(pf) |> AT
+    u_min, u_max, x_min, x_max, p_min, p_max = PS.get_bound_constraints(pf)
```

Comment:
Still need to work on the abstraction there. `PowerSystem` should not be aware of the indexing of `u` and `x` (as done here), as we want to implement different formulations based on `PowerSystem`